x-pack/legacy/plugins/rollup/public/plugin.tsx

```diff
@@ -0,0 +1,155 @@
+// @ts-ignore
+import { App } from './crud_app/app';
+import { ManagementStart } from '../../../../../src/plugins/management/public';
+// @ts-ignore
+import { rollupJobsStore } from './crud_app/store';
+import { KibanaContextProvider } from '../../../../../src/plugins/kibana_react/public';
+// @ts-ignore
+import { setEsBaseAndXPackBase, setHttp } from './crud_app/services';
+import { setNotifications, setFatalErrors } from './kibana_services';
+
+export interface RollupPluginSetupDependencies {
+  __LEGACY: {
+    aggTypeFilters: AggTypeFilters;
+    aggTypeFieldFilters: AggTypeFieldFilters;
+    editorConfigProviders: EditorConfigProviderRegistry;
+    addSearchStrategy: (searchStrategy: SearchStrategyProvider) => void;
+    management: ManagementSetup;
+    addBadgeExtension: (badgeExtension: any) => void;
+    addToggleExtension: (toggleExtension: any) => void;
+  };
+  home?: HomePublicPluginSetup;
+}
+
+export interface RollupPluginStartDependencies {
+  management: ManagementStart;
+}
+
+export class RollupPlugin implements Plugin {
+  setup(
+    core: CoreSetup,
+    {
+      __LEGACY: {
+        aggTypeFilters,
+        aggTypeFieldFilters,
+        editorConfigProviders,
+        addSearchStrategy,
+        management: managementLegacy,
+        addBadgeExtension,
+        addToggleExtension,
+      },
+      home,
+    }: RollupPluginSetupDependencies
+  ) {
+    setFatalErrors(core.fatalErrors);
+    addBadgeExtension(rollupBadgeExtension);
+    addToggleExtension(rollupToggleExtension);
+
+    const isRollupIndexPatternsEnabled = core.uiSettings.get(CONFIG_ROLLUPS);
+
+    if (isRollupIndexPatternsEnabled) {
+      managementLegacy.indexPattern.creation.add(RollupIndexPatternCreationConfig);
+      managementLegacy.indexPattern.list.add(RollupIndexPatternListConfig);
+      addSearchStrategy(getRollupSearchStrategy(core.http.fetch));
+      initAggTypeFilter(aggTypeFilters);
+      initAggTypeFieldFilter(aggTypeFieldFilters);
+      initEditorConfig(editorConfigProviders);
+    }
+
+    if (home) {
+      home.featureCatalogue.register({
+        id: 'rollup_jobs',
+        title: 'Rollups',
+        description: i18n.translate('xpack.rollupJobs.featureCatalogueDescription', {
+          defaultMessage:
+            'Summarize and store historical data in a smaller index for future analysis.',
+        }),
+        icon: 'indexRollupApp',
+        path: `#${CRUD_APP_BASE_PATH}/job_list`,
+        showOnHomePage: true,
+        category: FeatureCatalogueCategory.ADMIN,
+      });
+    }
+  }
+
+  start(core: CoreStart, { management }: RollupPluginStartDependencies) {
+    setHttp(core.http);
+    setNotifications(core.notifications);
+    setEsBaseAndXPackBase(core.docLinks.ELASTIC_WEBSITE_URL, core.docLinks.DOC_LINK_VERSION);
+
+    const esSection = management.sections.getSection('elasticsearch');
+
+    const I18nContext = core.i18n.Context;
+
+    esSection!.registerApp({
```

Comment:
I didn't actually know that we could register at plugin start time. This seems like something that could be done at setup time too, not sure rollup plugin has a start concern at the moment.

Reply:
The setup management API allows to register  an app, but for Rollup app we need `I18nContext` which is placed in CoreStart (`mount` for `registerApp` is sync function, do we cannot get CoreStart by calling `core.getStartServices()`)

Reply:
Hm, I see what you're saying. However, since the i18n start service is only used inside of the register app's `mount` function (which is `async`) you could call `core.getStartServices` there and `setup` it would remain sync. To be clear, I don't think this is a major issue.

Reply:
yeah, you're right, it's possible. Please take a look at the latest changes.